### PR TITLE
fix: use the cleaned up IPLD Format API

### PR DESCRIPTION
**Pull request by vmx**
BREAKING CHANGE: All formats now return data according to the [IPLD Data Model]

The most important change is that now *all* formats return links as [CID instances]
and no longer as the JSON representation (`{"/": "base-encoded-cid"}`.

[IPLD Data Model]: https://github.com/ipld/specs/blob/master/IPLD-Data-Model-v1.md
[CID instances]: https://github.com/multiformats/js-cid/

### CODE REVIEW

**Comment by rvagg** [comment]
I'll save a review until this is in sync with the latest format changes

**Comment by vmx** [comment]
@rvagg It's ready for review now.

File: test/ipld-all.js [hunk 1]
```diff
@@ -31,92 +29,61 @@ describe('IPLD Resolver for dag-cbor + dag-pb', () => {
...
-          }).then(
-            () => cb(null),
-            (error) => cb(error)
-          )
-        }, cb)
-      }
-    ], done)
+  before(async () => {
+    resolver = await new Promise((resolve, reject) => {
+      inMemory(IPLDResolver, (error, res) => {
```
Comment (rvagg) [inline test/ipld-all.js]: this looks easily promisifiable, up to you

File: test/ipld-all.js [hunk 2]
```diff
@@ -31,92 +29,61 @@ describe('IPLD Resolver for dag-cbor + dag-pb', () => {
...
+        if (error) reject(error)
+        else resolve(res)
+      })
+    })
+
+    nodePb = await dagPB.DAGNode.create(Buffer.from('I am inside a Protobuf'))
```
Comment (rvagg) [inline test/ipld-all.js]: this shouldn't be an async call now right?

Comment (rvagg) [reply test/ipld-all.js]: I see a few more of these below too

File: test/ipld-dag-cbor.js [hunk 3]
```diff
@@ -26,55 +25,32 @@ module.exports = (repo) => {
...
-          })
-        }
-      ], store)
-
-      async function store () {
-        const nodes = [node1, node2, node3]
-        const result = resolver.putMany(nodes, multicodec.DAG_CBOR)
-        ;[cid1, cid2, cid3] = await result.all()
+      node1 = { someData: 'I am 1' }
+      const serialized1 = await dagCBOR.util.serialize(node1)
```
Comment (rvagg) [inline test/ipld-dag-cbor.js]: not async now, right?

File: test/ipld-all.js (hunk 1 above)
Comment (vmx) [reply test/ipld-all.js]: Sadly isn't promisifiable. According to Alan because it thinks IPLDResolver is function, hence a callback.

**Comment by vmx** [comment]
@rvagg Thanks for being so thorough. I only looked at the source and not at the tests to see if there's former async left-overs.

**Comment by vmx** [comment]
This change got approved. Only change is a fixup + updating the IPLD Format implementations to the current versions so that things actually work. Hence merging.

**Pull request merged**